src/jit/lowerxarch.cpp

```diff
@@ -1110,6 +1116,23 @@ void Lowering::TreeNodeInfoInitCall(GenTreeCall* call)
         // computed into a register.
         if (!call->IsFastTailCall())
         {
+#ifdef _TARGET_X86_
+            // On x86, we need to generate a very specific pattern for indirect VSD calls:
+            //
+            //    3-byte nop
+            //    call dword ptr [eax]
+            //
+            // Where EAX is also used as an argument to the stub dispatch helper. Make
+            // sure that the call target address is computed into EAX in this case.
+            if (((call->gtFlags & GTF_CALL_VIRT_KIND_MASK) == GTF_CALL_VIRT_STUB) && (call->gtCallType == CT_INDIRECT))
+            {
+                assert(ctrlExpr->isIndir());
+
+                ctrlExpr->gtGetOp1()->gtLsraInfo.setDstCandidates(l, RBM_EAX);
```

Comment:
Just to be clear, we're currently calling `setDstCandidates` on the operand to the `ctrlExpr`. So given the IR in the example above, we're calling `setDstCandidates` on the node for `t70` (i.e. the use of `loc3`). Calling `setSrcCandidates` seems backwards to me (though it would seem to make sense if we were calling it on `ctrlExpr`).